src/devices/Seesaw/samples/Seesaw.Sample.Encoder.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using Iot.Device.Seesaw;
+
+using Seesaw seesawDevice = new Seesaw(I2cDevice.Create(new I2cConnectionSettings(1, 0x36)));
+
+seesawDevice.EnableEncoderInterrupt();
```

Comment:
I suppose the interrupt requires a GPIO pin? You are not using any GPIO pins in this example. Can you add an example to use the interrupt?

Reply:
For the implementation of the encoder functions I followed the documentation of Adafruit's Circuit Python. The python method  [enable_encoder_interrupt](https://docs.circuitpython.org/projects/seesaw/en/latest/api.html#adafruit_seesaw.seesaw.Seesaw.enable_encoder_interrupt) does not use any interrupt either.

I extended the seesaw class with the encoder functionality with the goal to talk to the [Adafruit I2C QT Rotary Encoder](https://learn.adafruit.com/adafruit-i2c-qt-rotary-encoder). It uses seesaw and returns changes of the rotary encoder. The interrupt to be enabled on the board is board specific and therefore implemented directly by the board (i.e. by the Adafruit I2C QT Rotary Encoder Board). With the enable_encoder_interrupt function I can enable it without knowing exactly which GPIO is used.

Reply:
@FloGs Sorry for being unclear. The `EnableEncoderInterrupt()` enables the interrupt _output_ of the device. That pin is fixed on the breakout (labeled "INT") and therefore doesn't need an argument. 

To use it though, you need a GpioController on the host (e.g. your Raspberry Pi) and an input pin configured for interrupt triggering there. So instead of polling for changes on the encoder (which is what your example currently does) you wait until the interrupt is raised and only then read the new value. That saves a lot of CPU time.

Reply:
@pgrawehr Thank you for the explanation! It seems that I had not fully understood what the EnableEncoderInterrupt is used for. I was so focused on Seesaw and its functions and therefore just implemented the functions that are also used in the Python library. I connected the board via STEMMA QT cable, and therefore overlooked the INT pin.

After your explanation I now understand what the method is used for. Of course it makes much more sense that way :sweat_smile: 

I will adapt my test setup accordingly and then modify the sample project to use GpioController.

Reply:
@pgrawehr The example project now uses interrupts both for the detection of the rotary encoder position and for the detection of a push switch change.